OSDOCS#7528: Reorganizing namespace exclusions and disabling after mo… (#64343)
openshift/openshift-docs: modules/security-context-constraints-psa-sync-exclusions.adoc

// Module included in the following assemblies:
//
// * authentication/understanding-and-managing-pod-security-admission.adoc
// * operators/operator_sdk/osdk-complying-with-psa.adoc

:_content-type: CONCEPT
[id="security-context-constraints-psa-sync-exclusions_{context}"]
= Pod security admission synchronization namespace exclusions

> Pod security admission synchronization is permanently disabled on most system-created namespaces. Synchronization is also initially disabled user-created `openshift-*` prefixed namespaces, but you can enable synchronization on them later.

Reviewer: ```suggestion
Pod security admission synchronization is permanently disabled on most system-created namespaces. Synchronization is also initially disabled on user-created `openshift-*` prefixed namespaces, but you can enable synchronization on them later.
```

[IMPORTANT]
====

> If the pod security admission labels (`pod-security.kubernetes.io/<mode>`) are manually modified away from the system defaults on a namespace, synchronization is disabled for that namespace. Users can enable synchronization again, if necessary.

Reviewer: See if the following makes sense
```suggestion
If a pod security admission label (`pod-security.kubernetes.io/<mode>`) is manually modified from the auto-labelled value on a label-synchronized namespace, synchronization is disabled for that label. Users can enable synchronization again, if necessary, by removing the label from the namespace or by forcing synchronization by setting the `security.openshift.io/scc.podSecurityLabelSync` label to `true`.
```

// TODO: Anything users need to know about re-enabling? Will anything be overwritten, etc.?
====

[discrete]
== Permanently disabled namespaces

Namespaces that are defined as part of the cluster payload have pod security admission synchronization disabled permanently. The following namespaces are permanently disabled:

* `default`
* `kube-node-lease`
* `kube-system`
* `kube-public`
* `openshift`
* All system-created namespaces that are prefixed with `openshift-`, except for `openshift-operators`

[discrete]
== Initially disabled namespaces

By default, all namespaces that have an `openshift-` prefix have pod security admission synchronization disabled initially. You can enable synchronization for user-created [x-]`openshift-*` namespaces and for `openshift-operators`.

[NOTE]
====
You cannot enable synchronization for any system-created [x-]`openshift-*` namespaces, except for `openshift-operators`.
====

> If an Operator is installed in a user-created `openshift-*` namespace, synchronization enabled automatically after a cluster service version (CSV) is created in the namespace. The synchronized label inherits the permissions of the service accounts in the namespace.

Reviewer: ```suggestion
If an Operator is installed in a user-created `openshift-*` namespace, synchronization is enabled automatically after a cluster service version (CSV) is created in the namespace. The synchronized label inherits the permissions of the service accounts in the namespace.
```

"the label inherits permissions" sounds odd, should we say that it's "derived from the permissions"?

// Module included in the following assemblies:
//
// * authentication/understanding-and-managing-pod-security-admission.adoc
// * operators/operator_sdk/osdk-complying-with-psa.adoc

:_content-type: CONCEPT
[id="security-context-constraints-psa-sync-exclusions_{context}"]
= Pod security admission synchronization namespace exclusions

Pod security admission synchronization is permanently disabled on most system-created namespaces. Synchronization is also initially disabled on user-created `openshift-*` prefixed namespaces, but you can enable synchronization on them later.

> [IMPORTANT]

Author: Yeah I played with this a few different ways, and wasn't super happy with any of them. This was the best way I could come up with, but I'm open to other ideas if you have them

> [IMPORTANT]

Reviewer 2: I also couldn't think of anything particularly helpful... maybe next time this module needs an edit something brilliant will occur :joy:

====
If a pod security admission label (`pod-security.kubernetes.io/<mode>`) is manually modified from the automatically labeled value on a label-synchronized namespace, synchronization is disabled for that label.

If necessary, you can enable synchronization again by using one of the following methods:

* By removing the modified pod security admission label from the namespace
* By setting the `security.openshift.io/scc.podSecurityLabelSync` label to `true`
+
If you force synchronization by adding this label, then any modified pod security admission labels will be overwritten.
====

[discrete]
== Permanently disabled namespaces

Namespaces that are defined as part of the cluster payload have pod security admission synchronization disabled permanently. The following namespaces are permanently disabled:

* `default`
* `kube-node-lease`
* `kube-system`
* `kube-public`
* `openshift`
* All system-created namespaces that are prefixed with `openshift-`, except for `openshift-operators`

[discrete]
== Initially disabled namespaces

> By default, all namespaces that have an `openshift-` prefix have pod security admission synchronization disabled initially. You can enable synchronization for user-created [x-]`openshift-*` namespaces and for `openshift-operators`.

Reviewer 2: ```suggestion
By default, all namespaces that have an `openshift-` prefix have pod security admission synchronization disabled initially. You can enable synchronization for user-created [x-]`openshift-*` namespaces and for the `openshift-operators` namespace.
```

[NOTE]
====

> You cannot enable synchronization for any system-created [x-]`openshift-*` namespaces, except for `openshift-operators`.

Reviewer 2: ```suggestion
You cannot enable synchronization for any system-created [x-]`openshift-*` namespaces, except for the `openshift-operators` namespace.
```

As an aside, what does `[x-]` do here?

Author: Sometimes when you have an asterisk inside backticks, it has issues in thinking you're trying to bold. I can't remember the exact scenarios when issues happens, but it sometimes fails on Travis builds and sometimes will bold text afterward that you didn't mean to. This `[x-]` escapes that so that it doesn't try to process the * as bold.

I can't tell for sure if it's needed in this case, but since it was there in the text I copied from, it's safest to leave it in.
